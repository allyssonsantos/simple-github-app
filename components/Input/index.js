import styled from 'styled-components';
import Theme from '../Theme';

const Input = styled.input.attrs({
  type: 'text',
  placeholder: 'Insert github username',
})`
  border: 1px solid ${Theme.colors.secondary};
  border-radius: 22px;
  font-size: ${props =>
    props.small ? Theme.font.size.small : Theme.font.size.medium};
  height: ${props => (props.small ? '14px' : '42px')};
  padding: ${Theme.spacing.padding.small};
`;

export default Input;
