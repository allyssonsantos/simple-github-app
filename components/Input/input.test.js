import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Input from './';

describe('With Snapshot Testing', () => {
  it('Should match snapshot without props', () => {
    const component = renderer.create(<Input />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match snapshot with small prop', () => {
    const component = renderer.create(<Input small />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
