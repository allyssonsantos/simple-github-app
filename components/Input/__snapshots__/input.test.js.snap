// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Should match snapshot with small prop 1`] = `
.c0 {
  border: 1px solid #999;
  border-radius: 22px;
  font-size: 16px;
  height: 14px;
  padding: 10px;
}

<input
  className="c0"
  placeholder="Insert github username"
  type="text"
/>
`;

exports[`With Snapshot Testing Should match snapshot without props 1`] = `
.c0 {
  border: 1px solid #999;
  border-radius: 22px;
  font-size: 18px;
  height: 42px;
  padding: 10px;
}

<input
  className="c0"
  placeholder="Insert github username"
  type="text"
/>
`;
